### PR TITLE
Miscellaneous bugfixes and code clean-up

### PR DESCRIPTION
**Pull request by vsreekanti**
* Adds configurable scheduling policies.
* Minor bugfixes introduced in the last PR (e.g., correctly dealing with results).

### CODE REVIEW

File: cloudburst/server/scheduler/policy/default_policy.py [hunk 1]
```diff
@@ -248,6 +265,13 @@ def pin_function(self, dag_name, function_ref, colocated):
             # pin this function there.
             node, tid = sys_random.sample(candidates, 1)[0]
 
+            for other in self.pending_dags[dag_name]:
+                _, thread = other
+                other_node, _ = other
```
Comment (cw75) [inline cloudburst/server/scheduler/policy/default_policy.py:270]: can these be combined into one line?

File: cloudburst/server/scheduler/policy/default_policy.py [hunk 2]
```diff
@@ -292,9 +316,9 @@ def pin_function(self, dag_name, function_ref, colocated):
     def commit_dag(self, dag_name):
         for function_name, location in self.pending_dags[dag_name]:
             if function_name not in self.function_locations:
-                self.function_locations[function_name] = set()
+                self.function_locations[function_name] = []
```
Comment (cw75) [inline cloudburst/server/scheduler/policy/default_policy.py:319]: why are we changing from set to list?

Comment (vsreekanti) [reply cloudburst/server/scheduler/policy/default_policy.py:319]: This allows us to do round-robin scheduling. `set` don't have order but `list`s do.

File: conf/cloudburst-local.yml [hunk 3]
```diff
@@ -7,6 +7,7 @@ executor:
 scheduler:
   routing_address: 127.0.0.1
   metric_address: 127.0.0.1
+  policy: locality
```
[the remark this answers is not in the document]
Comment (vsreekanti) [reply conf/cloudburst-local.yml:10]: We support three scheduling policies: locality-based (default), round-robin, and random.

File: cloudburst/server/scheduler/policy/default_policy.py (hunk 1 above)
Comment (vsreekanti) [reply cloudburst/server/scheduler/policy/default_policy.py]: Fixed.

File: cloudburst/server/scheduler/policy/default_policy.py (hunk 2 above)
Comment (cw75) [reply cloudburst/server/scheduler/policy/default_policy.py:319]: got it.

File: conf/cloudburst-local.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (cw75) [reply conf/cloudburst-local.yml:10]: ok

File: cloudburst/server/scheduler/policy/default_policy.py (hunk 1 above)
Comment (vsreekanti) [reply cloudburst/server/scheduler/policy/default_policy.py]: Fixed.

**Pull request merged**